example_clients/waveserv.py

```diff
@@ -0,0 +1,623 @@
+        vec = self.IUFind("trans_details")
+
+        value = float(value[0])
+        vec[item].value = value
+        self.IDSet(vec)
+
+
+    @MSG.subscribe("WAVESERV", 
+             "i230","i231","i232","i233","i263","i267",
+             "p205","p206","p207","p202","p203",
+             "i216","i217",
+             "p208",
+             "i211","i295","i215","i264","i265")
+    def on_trans_servo_change(self, item, value):
+        vec = self.IUFind("trans_servo")
+
+        value = float(value[0])
+        vec[item].value = value
+        self.IDSet(vec)
+
+
+    @MSG.subscribe("WAVESERV","fA","fC","fT",)
+    def on_focus_change(self, item, value ):
+        vec = self.IUFind("focus")
+        if item == "fA":
+            vec["actual"].value = float(value[0])
+        elif item == "fC":
+            vec["commanded"].value = float(value[0])
+        elif item == "fT":
+            vec["target"].value = float(value[0])
+   
+        self.IDSet(vec)
+
+
+    @MSG.subscribe("WAVESERV", 
+            "fE", "fMlimPos", "fplimPos", "fHomePos","fDAC", "fDACBias")
+    def on_focus_details_change(self, item, value):
+        vec = self.IUFind("focus_details")
+
+        value = float(value[0])
+        vec[item].value = value
+        self.IDSet(vec)
+
+
+    @MSG.subscribe("WAVESERV", 
+             "i430","i431","i432","i433","i463","i467",
+             "p405","p406","p407","p402","p403",
+             "i416","i417",
+             "p408",
+             "i411","i495","i415","i464","i465")
+    def on_focus_servo_change(self, item, value):
+        vec = self.IUFind("focus_servo")
+
+        value = float(value[0])
+        vec[item].value = value
+        self.IDSet(vec)
+
+
+
+    @MSG.subscribe("WAVESERV","cA","cC","ct",)
+    def on_camera_change(self, item, value ):
+        vec = self.IUFind("select")
+        if item == "cA":
+            vec["actual"].value = float(value[0])
+        elif item == "cC":
+            vec["commanded"].value = float(value[0])
+        elif item == "ct":
+            vec["target"].value = float(value[0])
+   
+        self.IDSet(vec)
+
+
+    @MSG.subscribe("WAVESERV", 
+            "cE", "cMlimPos", "cplimPos", "cHomePos","cDAC", "cDACBias")
+    def on_camera_details_change(self, item, value):
+        vec = self.IUFind("camera_details")
+
+        value = float(value[0])
+        vec[item].value = value
+        self.IDSet(vec)
+
+
+    @MSG.subscribe("WAVESERV", 
+             "i330","i331","i332","i333","i363","i367",
+             "p305","p306","p307","p302","p303",
+             "i316","i317",
+             "p308",
+             "i311","i395","i315","i364","i365")
+    def on_camera_servo_change(self, item, value):
+        vec = self.IUFind("camera_servo")
+
+        value = float(value[0])
+        vec[item].value = value
+        self.IDSet(vec)
+
+
+
+    @MSG.subscribe("WAVESERV","pA", "pt")
+    def on_puntino_change(self, item, value ):
+        vec = self.IUFind("puntino")
```

Comment:
fwiw, the puntino was removed from the WFS over 15 years ago and replaced with the stellacam.